src/plugins/plugin-hapi.ts

```diff
@@ -34,87 +42,121 @@ function getFirstHeader(req: IncomingMessage, key: string): string|null {
+    root.addLabel(api.labels.HTTP_URL_LABEL_KEY, url);
+    root.addLabel(api.labels.HTTP_SOURCE_IP, req.connection.remoteAddress);
+
+    // wrap end
+    res.end = function(this: ServerResponse) {
+      res.end = originalEnd;
+      const returned = res.end.apply(this, arguments);
+
+      if (request.route && request.route.path) {
+        root.addLabel('hapi/request.route.path', request.route.path);
       }
+      root.addLabel(api.labels.HTTP_RESPONSE_CODE_LABEL_KEY, res.statusCode);
+      root.endSpan();
 
-      api.wrapEmitter(req);
-      api.wrapEmitter(res);
-
-      const url = `${req.headers['X-Forwarded-Proto'] || 'http'}://${
-          req.headers.host}${req.url}`;
-
-      // we use the path part of the url as the span name and add the full
-      // url as a label
-      // req.path would be more desirable but is not set at the time our
-      // middleware runs.
-      root.addLabel(api.labels.HTTP_METHOD_LABEL_KEY, req.method);
-      root.addLabel(api.labels.HTTP_URL_LABEL_KEY, url);
-      root.addLabel(api.labels.HTTP_SOURCE_IP, req.connection.remoteAddress);
-
-      // wrap end
-      res.end = function(this: ServerResponse) {
-        res.end = originalEnd;
-        const returned = res.end.apply(this, arguments);
-
-        if (request.route && request.route.path) {
-          root.addLabel('hapi/request.route.path', request.route.path);
-        }
-        root.addLabel(api.labels.HTTP_RESPONSE_CODE_LABEL_KEY, res.statusCode);
-        root.endSpan();
-
-        return returned;
-      };
-
-      // if the event is aborted, end the span (as res.end will not be called)
-      req.once('aborted', () => {
-        root.addLabel(api.labels.ERROR_DETAILS_NAME, 'aborted');
-        root.addLabel(
-            api.labels.ERROR_DETAILS_MESSAGE, 'client aborted the request');
-        root.endSpan();
-      });
+      return returned;
+    };
 
-      return reply.continue();
+    // if the event is aborted, end the span (as res.end will not be called)
+    req.once('aborted', () => {
+      root.addLabel(api.labels.ERROR_DETAILS_NAME, 'aborted');
+      root.addLabel(
+          api.labels.ERROR_DETAILS_MESSAGE, 'client aborted the request');
+      root.endSpan();
     });
-  };
+
+    return continueCb();
+  });
 }
 
-const plugin: PluginTypes.Plugin = [{
-  file: '',
-  versions: SUPPORTED_VERSIONS,
-  patch: (hapi, api) => {
-    shimmer.wrap(hapi.Server.prototype, 'connection', (connection) => {
-      return function connectionTrace(this: {}) {
-        const server: hapi_16.Server = connection.apply(this, arguments);
-        server.ext('onRequest', createMiddleware(api));
-        return server;
-      };
-    });
-  },
-  unpatch: (hapi) => {
-    shimmer.unwrap(hapi.Server.prototype, 'connection');
-  }
-} as PluginTypes.Patch<Hapi16Module>];
+const plugin: PluginTypes.Plugin = [
+  {
+    versions: '8 - 16',
+    patch(hapi, api) {
+      function createMiddleware(): hapi_16.ServerExtRequestHandler {
```

Comment:
This can be an arrow function, no?

Reply:
Yep, done

Reply:
What I asked was about `createMiddleware()` :) But `patch` and `unpatch` also seem better as arrow functions.

Reply:
Ah, gotcha -- done (x2)